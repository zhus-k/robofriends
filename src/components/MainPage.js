import React, { Component } from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import Header from '../components/Header';
import ErrorBoundary from '../components/ErrorBoundary';
import '../containers/App.css';
import 'tachyons';

class MainPage extends Component {

    componentDidMount() {
        this.props.onRequestRobots();
    }

    filteredRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        });
    }

    render() {
        const {onSearchChange, isPending } = this.props;

        return isPending ?
            <h1 className='tc'>Loading...</h1>
            :
            (
                <div className='tc'>
                    <Header />
                    <div className='divider'></div>
                    <SearchBox searchChange={ onSearchChange } />
                    <Scroll>
                        <ErrorBoundary>
                            <CardList robots={ this.filteredRobots() } />
                        </ErrorBoundary>
                    </Scroll>
                </div>
            );
    };
};

export default MainPage;