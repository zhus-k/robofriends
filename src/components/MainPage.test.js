import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }

    wrapper = shallow(<MainPage {...mockProps} />);
});

it('expect App to render', () => {
    expect(wrapper).toMatchSnapshot();
});

it('expect pending', () => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: true
    }
    const wrapper = shallow(<MainPage {...mockProps} />)
    expect(wrapper.props()).toMatchSnapshot();
})

it('filters robots correctly', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3000,
            name: 'john',
            email: 'jd@email.com'
        }],
        searchField: 'john',
        isPending: false
    }
    const wrapper2 = shallow(<MainPage {...mockProps2} />);

    expect(wrapper2.instance().filteredRobots()).toEqual([
        {
            id: 3000,
            name: 'john',
            email: 'jd@email.com'
        }
    ]);
});

it('filters robots correctly 2', () => {
    const mockProps3 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3000,
            name: 'john',
            email: 'jd@email.com'
        }],
        searchField: 'a',
        isPending: false
    }

    const filteredRobots = [];
    const wrapper3 = shallow(<MainPage {...mockProps3} />);

    expect(wrapper3.instance().filteredRobots()).toEqual(filteredRobots);
});