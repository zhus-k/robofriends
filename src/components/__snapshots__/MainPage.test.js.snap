// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect App to render 1`] = `ShallowWrapper {}`;

exports[`expect pending 1`] = `
Object {
  "children": "Loading...",
  "className": "tc",
}
`;
